packages/boot-exec-derive/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+#![recursion_limit = "128"]
+
+extern crate proc_macro;
+use convert_case::{Case, Casing};
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, DeriveInput, Fields, Ident};
+
+#[proc_macro_derive(ExecuteFns,attributes(payable))]
+pub fn boot_execute(input: TokenStream) -> TokenStream {
```

Comment:
Will this be able to handle more advanced top-level structs like the `Middleware` ones with generics?

Reply:
Nope this is just for regular contracts. We could augment it to fit our needs in the Abstract crates.